replay/obp_evaluation/replay_offline.py

```diff
@@ -0,0 +1,235 @@
+import numpy as np
+import pandas as pd
+import time
+import logging
+
+from dataclasses import dataclass
+
+from obp.policy.base import BaseOfflinePolicyLearner
+from obp.policy.policy_type import PolicyType
+from replay.utils import convert2spark
+from replay.data_preparator import DataPreparator, Indexer
+from pyspark.sql import DataFrame
+from replay.obp_evaluation.obp_optuna_objective import OBPObjective
+from replay.obp_evaluation.utils import split_bandit_feedback
+from replay.models.base_rec import BaseRecommender
+
+from optuna import create_study
+from optuna.samplers import TPESampler
+
+from typing import (
+    Any,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Union,
+    Sequence,
+    Set,
+    Tuple,
+)
+
+def obp2df(action: np.ndarray,
+           reward: np.ndarray,
+           timestamp: np.ndarray) -> Optional[pd.DataFrame]:
+    """
+        Converts OBP log to the pandas DataFrame
+    """
+
+    n_interactions = len(action)
+
+    d = {
+        "user_idx": np.arange(n_interactions),
+        "item_idx": action,
+        "relevance": reward,
+        "timestamp": timestamp,
+    }
+    df = pd.DataFrame(d)
+
+    return df
+
+def context2df(context: np.ndarray,
+               idx_col: np.ndarray,
+               idx_col_name: str) -> Optional[pd.DataFrame]:
+    """
+        Converts OBP log to the pandas DataFrame
+    """
+
+    df1 = pd.DataFrame({
+        idx_col_name+"_idx": idx_col
+    })
+    df2 = pd.DataFrame(context, columns=[str(i)+"_"+idx_col_name for i in range(context.shape[1])])
+
+    return df1.join(df2)
+
+@dataclass
+class RePlayOfflinePolicyLearner(BaseOfflinePolicyLearner):
```

Comment:
Why don't you inherit `RePlayOfflinePolicyLearner` from both `BaseOfflinePolicyLearner` and `BaseRecommender`?

Reply:
I can't see if it simplifies the overall logic. I see that it helps to remove `logger()` function, but it seems that I should hold `replay_model` inside class anyway. Can you comment more on that?